Source/FieldSolver/SpectralSolver/SpectralAlgorithms/ComovingPsatdAlgorithm.cpp

```diff
@@ -64,6 +66,8 @@ ComovingPsatdAlgorithm::ComovingPsatdAlgorithm (const SpectralKSpace& spectral_k
 void
 ComovingPsatdAlgorithm::pushSpectralFields (SpectralFieldData& f) const
 {
+    const SpectralFieldIndex& Idx = m_spectral_index;
```

Comment:
I believe it will copy the referenced type by value - at least, that is my parsing of the first answer here: https://stackoverflow.com/questions/19676473/in-a-lambda-how-reference-is-being-captured-by-value

Reply:
Yes, it will copy `Idx` by value to device.

> otherwise this would capture and copy the whole instance of ComovingPsatdAlgorithm (of which m_spectral_index is a member) to the device.

We don't want to use `m_spectral_index`, but it's for a different reason.  `m_spectral_index` is really `this->m_spectral_index`.  So the lambda captures the pointer `this`.  This is an efficient way on host, but wrong for device code because `*this` does not live in device memory.

Reply:
Great! Thanks for your answers!